app/models/summoner.rb

```diff
@@ -5,13 +5,25 @@ class Summoner < ActiveRecord::Base
   validates :riot_id, presence: true, uniqueness: true
   validates :name, presence: true
 
-  def self.add id
-    if Summoner.exists?(riot_id: id)
-      return Summoner.find_by(riot_id: id)
-    else
-      summoner = Riot.summoner_id(id)
-      summoner = summoner[id.to_s]
-      return Summoner.new(riot_id: summoner["id"], name: summoner["name"])
+  def self.add id_array
```

Comment:
Assignment Branch Condition size for add is too high. [17.72/15]<br>Use def with parentheses when there are parameters.